connect-button/src/main/java/com/ifttt/connect/ui/BaseConnectButton.java

```diff
@@ -815,15 +832,55 @@ public void onAnimationEnd(Animator animation) {
                             }
 
                             dispatchState(Login);
-                            buttonApiHelper.connect(getContext(), connection, emailEdt.getText().toString(),
-                                    buttonState);
+                            buttonApiHelper.connect(getContext(),
+                                connection,
+                                emailEdt.getText().toString(),
+                                buttonState
+                            );
                             monitorRedirect();
                         }
                     });
 
                     completeProgress.start();
                 }
             }
+        };
+
+        CharSequence text = getResources().getString(R.string.connecting);
+        Animator showProgress = progressView.progress(0f, 0.5f, text, ANIM_DURATION_LONG);
+        showProgress.setInterpolator(LINEAR_INTERPOLATOR);
+        showProgress.addListener(new CancelAnimatorListenerAdapter(animatorLifecycleObserver) {
+
+            @Override
+            public void onAnimationStart(Animator animation) {
+                super.onAnimationStart(animation);
+                if (isCanceled()) {
+                    return;
+                }
+
+                // Remove icon elevation when the progress bar is visible.
+                ViewCompat.setElevation(iconImg, 0f);
+
+                buttonApiHelper.prepareAuthentication(emailEdt.getText().toString(), () -> {
```

Comment:
This and the `onAuthPreparedAnimatorListener` is the main changes here, basically we are checking the animation running status when the AsyncTask completes:
* if the animation is still running: we add the listener to the animation, which has the logic to continue the flow
* if the animation is not running: we invoke the listener method directly.

Sorry for the extra style changes, I meant to do it in a separate PR for consistency but ended up just applying it here.